rubi/rubi/data/market.py

```diff
@@ -445,14 +446,16 @@ def _build_offers_query(
         ]
         where = [condition for condition in where if condition is not None]
 
-        offers_query = self.subgraph.Query.offers(  # noqa
+        offers_field_path = self.subgraph.Query.offers(  # noqa
             orderBy=order_by,
             orderDirection=order_direction,
             first=first,
             where=where if where else {},
         )
 
-        return offers_query
+        block_field_path = self.subgraph.Query._meta().block()  # noqa
+
+        return {"offer": offers_field_path, "block": block_field_path}
 
     def _build_trades_query(
```

Comment:
i think it makes sense to also update the way we retrieve trades to match what we are doing with `offers` in this PR, it seems like a better approach going forward